pywt/src/_pywt.pyx

```diff
@@ -790,30 +796,43 @@ def idwt(cA, cD, object wavelet, object mode='sym', int correct_size=0):
     """
     # accept array_like input; make a copy to ensure a contiguous array
 
-    if cA is None and cD is None:
+    has_cA = not (cA is None)
+    has_cD = not (cD is None)
+    if not (has_cA or has_cD):
         raise ValueError("At least one coefficient parameter must be "
                          "specified.")
 
-    if cA is not None:
+    # for complex inputs: compute real and imaginary separately then combine
+    if (has_cA and np.iscomplexobj(cA)) or (has_cD and np.iscomplexobj(cD)):
+        if not has_cA:
+            cD = np.asarray(cD)
+            cA = np.zeros_like(cD)
+        elif not has_cD:
+            cA = np.asarray(cA)
+            cD = np.zeros_like(cA)
+        return (idwt(cA.real, cD.real, wavelet, mode, correct_size) +
+                1j*idwt(cA.imag, cD.imag, wavelet, mode, correct_size))
+
+    if has_cA:
         dt = _check_dtype(cA)
         cA = np.array(cA, dtype=dt)
         if cA.ndim != 1:
             raise ValueError("idwt requires 1D coefficient arrays.")
-    if cD is not None:
```

Comment:
I don't see the need to replace all these - `cD is not None` isn't much longer and shows the intention a bit more immediately, whereas otherwise I'd have to go back and check what `has_cD` means. Just my personal preference though, both valid.